### PR TITLE
Fix/Update Documentation: Bots

### PR DESCRIPTION
**Pull request by bandantonio**
- Revised Bots README page to be more consistent and organized
- Renamed README page from "Using Bots" to "Bots Overview"
- Deleted Bots FAQ page as redundant

### CODE REVIEW

File: bots/README.md [hunk 1]
```diff
@@ -1,41 +1,69 @@
...
-To contribute to features under development see our
-[Bots Project issues](https://github.com/RocketChat/Rocket.Chat/projects/16).
+Rocket.Chat's framework (Meteor) has built-in utilities for streaming updates to data collections.
+Every user in Rocket.Chat has such a collection for their messages. Bots subscribe to
+the collection of messages for their user account, creating a stream that is updated every time a message is sent either directly to them or any room they are joined in.
+
+For more details, see the [Bots Architecture section](bots-architecture/).
+
+## Bots types
+
```
Comment (Sing-Li) [inline bots/README.md]: Please rework this section.  We do not support internal BOTs at all moving forward.

Perhaps it is good to make a "minor reference" to distinguish between BOTs (always running externally) and Rocket.Chat Apps (always running internally in the server).  Rocket.Chat Apps are in-server extensions or add-on,  designed to add features to the server; and almost NEVER have a conversational interface.

File: bots/README.md [hunk 2]
```diff
@@ -1,41 +1,69 @@
...
+[bBot](http://bbot.chat/), [Hubot](https://hubot.github.com/), [Botkit](https://botkit.ai/),
+[Botpress](https://botpress.io/).
+
+## How are bots hosted?
+
+The hosting for **external** bots depends on the platform. For example, Hubot can be
+launched on any Node.js environment, but you have to set it up yourself (using Heroku or
+Glitch services). Another bot platforms like Botkit provide their own services for hosting and provisioning bots. They just need a configuration to connect to your Rocket.Chat instance.
+
+**Internal** bots are hosted as part of the Rocket.Chat instance, either on-premise
```
Comment (Sing-Li) [inline bots/README.md]: again. please remove

File: bots/README.md [hunk 3]
```diff
@@ -1,41 +1,69 @@
...
+4. The bot's behaviour is defined via scripts according to the requirements of your framework. For example, check the details on how to [run a bBot bot](running-a-bbot-bot/).
 
-1. A bot user must be [created by an admin](creating-bot-users/) on the server
-2. The bot is run as [a separate process](bots-architecture/) using your chosen framework or platform
-3. The bot environment must be pre-configured by [setting environment variables](configure-bot-environment/)
-4. You provide scripts to define the bot's behaviour according to the requirements of your framework
-    - e.g. See [Running a bBot bot](running-a-bbot-bot/)
+## Next steps
+
+Get yourself familiar with [Bots Architecture](bots-architecture/).
```
Comment (Sing-Li) [inline bots/README.md:45]: a few simple diagrams and accompanying short description will REALLY HELP at this point.

External BOT architecture only.

Some attention should be paid to, in the architecture, -->  HOW DOES A BOT scale?   For example how can a BOT handle 1000 users concurrently.

**Review by Sing-Li — CHANGES_REQUESTED** [review]

**Comment by Sing-Li** [comment]
@bandantonio   Awesome work overall !   Please make the requested minor changes asap.

File: bots/README.md (hunk 1 above)
Comment (bandantonio) [reply bots/README.md]: Fixed this mentions as discussed

File: bots/README.md (hunk 2 above)
Comment (bandantonio) [reply bots/README.md]: Removed

File: bots/README.md (hunk 3 above)
Comment (bandantonio) [reply bots/README.md:45]: Let's leave this part as is at the moment. The architecture section will be reworked in the next PR

**Comment by bandantonio** [comment]
> @bandantonio Awesome work overall ! Please make the requested minor changes asap.

I fixed all your mentions. Please review the updated doc

**Review by Sing-Li — APPROVED** [review]
edited to remove explicit mention of "external bots" since it infers there may be other non-external ones.

edited to remove bot types section.  since there is  external vs other typing with Rocket.Chat.

**Pull request merged**